centreon/www/front_src/src/CloudNotificationsConfiguration/Actions/useDelete.tsx

```diff
@@ -0,0 +1,65 @@
+import { useAtom, useAtomValue, useSetAtom } from 'jotai';
+
+import {
+  deleteNotificationAtom,
+  isDeleteDialogOpenAtom,
+  isPanelOpenAtom,
+  selectedRowsAtom
+} from '../atom';
+import useDeleteRequest from '../api/useDeleteRequest';
+import { DeleteNotificationType } from '../models';
+
+interface UseDeleteState {
+  closeDialog: () => void;
+  deleteItems: ({ id, name, type }: DeleteNotificationType) => void;
+  isDialogOpen: boolean;
+  isLoading: boolean;
+  notificationName?: string;
+  openDialog: () => void;
+  submit: () => void;
+}
+
+const useDelete = (): UseDeleteState => {
+  const [isDialogOpen, setIsDialogOpen] = useAtom(isDeleteDialogOpenAtom);
+  const [deleteNotification, setDeleteInformations] = useAtom(
+    deleteNotificationAtom
+  );
+  const selectedRows = useAtomValue(selectedRowsAtom);
+  const setIsPanelOpen = useSetAtom(isPanelOpenAtom);
```

Comment:
I encountered an issue when using useState instead of useAtom. In my specific scenario, the dialog component failed to re-render when the state changed upon button click. However, after switching to an atom and utilizing useAtom, the component re-rendered as expected whenever the associated atom's value changed. While I'm not entirely certain about the underlying mechanics, it appears that the Jotai provider triggers component re-rendering if an atom subscribed to the component undergoes a change. @Thebarda